pkg/noderesources/types.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package noderesources
+
+import (
+	"bytes"
+	"encoding/json"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubernetes/pkg/scheduler/apis/config"
```

Comment:
Not sure I follow how to use in this case.

Reply:
I meant that we should use the "versioned" `ResourceSpec` from importing "k8s.io/kube-scheduler/config/v1": https://github.com/kubernetes/kube-scheduler/blob/master/config/v1/types.go#L144, instead of importing an internal (unversioned) `ResourceSpec` from "k8s.io/kubernetes/pkg/scheduler/apis/config".